keras_retinanet/bin/train.py

```diff
@@ -188,14 +191,19 @@ def parse_args():
     csv_parser.add_argument('classes', help='Path to a CSV file containing class label mapping.')
     csv_parser.add_argument('--val-annotations', help='Path to CSV file containing annotations for validation (optional).')
 
-    parser.add_argument('--weights', help='Weights to use for initialization (defaults to ImageNet).', default='imagenet')
-    parser.add_argument('--batch-size', help='Size of the batches.', default=1, type=int)
-    parser.add_argument('--gpu', help='Id of the GPU to use (as reported by nvidia-smi).')
-    parser.add_argument('--multi-gpu', help='Number of GPUs to use for parallel processing.', type=int, default=0)
+    parser.add_argument('--weights',       help='Weights to use for initialization (defaults to ImageNet).', default='imagenet')
+    parser.add_argument('--batch-size',    help='Size of the batches.', default=1, type=int)
+    parser.add_argument('--gpu',           help='Id of the GPU to use (as reported by nvidia-smi).')
+    parser.add_argument('--multi-gpu',     help='Number of GPUs to use for parallel processing.', type=int, default=0)
+    parser.add_argument('--epochs',        help='Number of epochs to train.', type=int, default=50)
+    parser.add_argument('--steps',         help='Number of steps per epoch.', type=int, default=10000)
+    parser.add_argument('--no-evaluation', help='Disable per epoch evaluation.', dest='evaluation', action='store_false')
```

Comment:
`--no-evaluation` is kinda double (although inverted) with `--val-annotations` for the `CSVGenerator`. We could instead add `--use-validation` to the COCO subparser to avoid having double options.

Reply:
Hmm I get it, but I would prefer that for COCO at least the evaluation is on by default. How would you suggest to do this?

Reply:
You can set mutually exclusive command line args by creating a group under parser.
Eg ```python
parser = argparse.ArgumentParser(prog='PROG')
group = parser.add_mutually_exclusive_group()
group.add_argument('--foo', action='store_true')
group.add_argument('--bar', action='store_false')```

Reply:
we could simply add `--no-validation` to the Coco subparser instead of the main parser I think.

Reply:
Named it `--no-evaluation`, but yeah.

When all datasets get evaluation code we can change this, for now this will do.

Reply:
`--val-annotations` and `--no-evaluation` isn't very consistent. I think we should use the same term for all generators.

If that term is "evaluation", how about changing `--val-annotations` to simply `--evaluate`? The description can make it clear that the argument should be a CSV annotations file.